src/t_hash.c

```diff
@@ -42,6 +42,11 @@ void hashTypeTryConversion(robj *o, robj **argv, int start, int end) {
     size_t sum = 0;
 
     if (o->encoding != OBJ_ENCODING_LISTPACK) return;
+    size_t args_length = end - start + 1;
+    if (args_length > server.hash_max_listpack_entries) {
```

Comment:
`hash_max_listpack_entries` should correspond to the length of hash not the length of listpack.
```suggestion
    if (args_length/2 > server.hash_max_listpack_entries) {
```

Reply:
What if duplicated field names are provided? Like `HSET key f 1 f 2 f 3 f 4 f 5 ...`. That could that result in a hash that is dict-encoded even though it small enough to fit in a listpack. Is that a problem?

Reply:
>  That could that result in a hash that is dict-encoded even though it small enough to fit in a listpack. Is that a problem?

It could, but that seems like an anti-pattern I don't care about.